api/azure-pipelines.ts

```diff
@@ -1,42 +1,130 @@
-import cheerio from 'cheerio'
 import got from '../libs/got'
 import { createBadgenHandler, PathArgs } from '../libs/create-badgen-handler'
+import { isNullOrUndefined } from 'util'
 
 export default createBadgenHandler({
   title: 'Azure Piplines',
   examples: {
-    '/azure-pipelines/yarnpkg/yarn/Yarn Acceptance Tests': 'build',
-    '/azure-pipelines/yarnpkg/yarn/Yarn Acceptance Tests/azure-pipelines': 'build (branch)',
+    '/azure-pipelines/build/status/yarnpkg/yarn/Yarn Acceptance Tests': 'build',
+    '/azure-pipelines/build/status/yarnpkg/yarn/Yarn Acceptance Tests/azure-pipelines': 'build (branch)',
   },
   handlers: {
-    '/azure-pipelines/:org/:project/:definition/:branch?': handler
+    '/azure-pipelines/:org/:project/:definition/:branch?': buildStatus,
+    '/azure-pipelines/build/status/:org/:project/:definition/:branch?': buildStatus,
+    '/azure-pipelines/build/version/:org/:project/:definition/:branch?': buildVersion,
+    '/azure-pipelines/build/test/:org/:project/:definition/:branch?': buildTestResult,
+    '/azure-pipelines/release/version/:org/:project/:definition?': releaseVersion,
+    '/azure-pipelines/deployment/version/:org/:project/:definition/:environment?': deployedReleaseVersion,
   }
 })
 
-async function handler ({ org, project, definition, branch = 'master'}: PathArgs) {
-  // @ts-ignore
-  const response = await got(`https://dev.azure.com/${org}/${project}/_apis/build/status/${definition}?branchName=${branch}`, { json: false })
-  const contentType = response.headers['content-type']
-
-  if (!contentType.includes('image/svg+xml')) {
-    return {
-      subject: 'Azure Pipelines',
-      status: 'unknown',
-      color: 'grey'
-    }
+const AZURE_DEVOPS_TOKEN = process.env.AZURE_DEVOPS_TOKEN
```

Comment:
Need to add this to [`now.json`](https://zeit.co/docs/configuration/#project/env):

```diff
	{
	  "env": {
	    "GH_TOKENS": "@badgen-gh-tokens",
	    "SENTRY_DSN": "@badgen-sentry-dsn",
	    "TRACKING_GA": "@badgen-tracking-ga",
+	    "AZURE_DEVOPS_TOKEN": "@badgen-azure-token"
	  }
	}
```